common/djangoapps/student/tests/tests.py

```diff
@@ -106,20 +106,40 @@ def test_cert_info(self):
         )
 
         cert_status = {'status': 'generating', 'grade': '67', 'mode': 'honor'}
-        self.assertEqual(
-            _cert_info(user, course, cert_status, course_mode),
-            {
-                'status': 'generating',
-                'show_disabled_download_button': True,
-                'show_download_url': False,
-                'show_survey_button': True,
-                'survey_url': survey_url,
-                'grade': '67',
-                'mode': 'honor',
-                'linked_in_url': None,
-                'can_unenroll': False,
-            }
-        )
+        with patch('lms.djangoapps.grades.new.course_grade.CourseGradeFactory.get_saved_grade') as patch_saved_grade:
+            patch_saved_grade.return_value = Mock(percent=100)
```

Comment:
This patch is what ended up solving the circular dependency issues I had been mentioning in standup. Simply mocking out the return value tests the logic we want to test, and avoids me needing to update this test to use a proper course instead of a `Mock`, which means we don't have to run the `CourseGradeFactory` code at all.